File: app/models/xy_master/chart_scope_info.rb
module XyMaster
  class ChartScopeInfo
    include ApplicationMemoryRecord
    memory_record [
      { key: "chart_scope_recently", name: "最近", date_gteq: 30.days, },
      { key: "chart_scope_all",      name: "全体", date_gteq: nil,     },
    ]
  end
end

File: nuxt_side/components/XyMaster/models/chart_scope_info.js
import { ApplicationMemoryRecord } from "@/components/models/application_memory_record.js"

export class ChartScopeInfo extends ApplicationMemoryRecord {
  static get define() {
    return [
      { key: "chart_scope_recently", name: "最近", },
      { key: "chart_scope_all",      name: "全体", },
    ]
  }
}
